### PR TITLE
Modified projection method

### PR DESCRIPTION
**Pull request by austinTalbot7241993**
**Description**
This pull request introduces significant enhancements to the spectral adaptation module, improving its robustness and efficacy in domain adaptation tasks. Specifically, it includes regularization of the covariance matrix estimation during the projection of new data into the learned "median" space. Additionally, the PR implements several more principled methods of "averaging" positive definite matrices, integrating them into the RotationAdaptation object for domain adaptation purposes.

**Key Enhancements**
1. Regularized Covariance Matrix Estimation: The PR adds regularization to the covariance matrix estimation process, enhancing the stability and reliability of projections into the learned "median" space. This regularization mitigates the effects of noise and uncertainty, resulting in more accurate adaptations.
2. Improved Matrix Averaging Methods: Several more principled means of "averaging" positive definite matrices are implemented in this PR. These methods offer a more robust approach to combining matrices, improving the accuracy and effectiveness of domain adaptation.
3. Integration into RotationAdaptation Object: The newly implemented matrix averaging methods are seamlessly incorporated into the RotationAdaptation object. This integration enhances the adaptability and versatility of the spectral adaptation module for domain adaptation tasks

### CODE REVIEW

File: python/python/bystro/covariance/tests/test_positive_definite_average.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    with pytest.raises(ValueError):
+        pd_mean_harmonic([])  # Empty list
+    with pytest.raises(ValueError):
+        pd_mean_harmonic(
+            [A, np.array([[1, 2], [2, 1]])]
+        )  # Not all positive definite
+
+
+def test_pd_mean_karcher():
```
Comment (akotlar) [inline python/python/bystro/covariance/tests/test_positive_definite_average.py:61]: could we add tests for the other euclidean and Riemann?

**Review by akotlar — CHANGES_REQUESTED** [review]
Let's add tests for the Riemann and Euclidean matrix means and then this is good to merge

**Review by akotlar — CHANGES_REQUESTED** [review]
Let's add tests for euclidian and riemann

**Comment by austinTalbot7241993** [comment]
Just removed, will add later.

**Review by akotlar — APPROVED** [review]
LGTM!

**Pull request merged**